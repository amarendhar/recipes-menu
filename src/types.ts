export type Maybe<T> = T | null;

export type Recipe = {
  id: string;
  title: string;
  image: string;
  tags: string[];
  description: string;
  chefName: string;
};
