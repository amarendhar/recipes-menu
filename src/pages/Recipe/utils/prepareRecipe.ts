import { EntryFields, EntrySkeletonType } from "contentful";
import { Maybe, RecipeEntry, RecipeData } from "types";

type RecipeFields = {
  title: EntryFields.Text;
  photo: EntryFields.AssetLink;
  tags: EntryFields.Array<EntryFields.EntryLink<EntrySkeletonType>>;
  description: EntryFields.Text;
  chef: EntryFields.EntryLink<EntrySkeletonType>;
};

export const prepareRecipe = (data: Maybe<RecipeEntry>): Maybe<RecipeData> => {
  if (!data) {
    return null;
  }

  const fields = data?.fields as RecipeFields;

  const id: string = data.sys.id;
  const title = fields?.title || "";
  const image = (fields?.photo?.fields?.file?.url || "") as string;
  const tags =
    fields?.tags?.map((tag) => (tag?.fields?.name as string) || "") || [];
  const description = fields?.description || "";
  const chef = (fields?.chef?.fields?.name || "") as string;
  const rating = 0;

  const recipeData: RecipeData = {
    id,
    title,
    image,
    tags,
    description,
    chef,
    rating,
  };

  return recipeData;
};
