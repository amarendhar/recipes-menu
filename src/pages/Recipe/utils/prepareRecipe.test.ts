import { prepareRecipe } from "./prepareRecipe";
import { RecipeData } from "types";
import mockRecipes from "mocks/__fixtures__/mockRecipes.json";

describe("# prepareRecipe", () => {
  it("Should return prepared recipe", () => {
    const recipe = prepareRecipe(mockRecipes.items[3] as RecipeData);

    expect(recipe).toEqual({
      id: expect.any(String),
      title: expect.any(String),
      image: expect.any(String),
      tags: expect.any(Array<String>),
      description: expect.any(String),
      chef: expect.any(String),
    });

    expect(recipe).toMatchSnapshot();
  });
});
