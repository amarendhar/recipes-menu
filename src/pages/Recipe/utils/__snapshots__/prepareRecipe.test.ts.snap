// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# prepareRecipe Should return prepared recipe 1`] = `
Object {
  "chef": "Jony Chives",
  "description": "Crispy chicken skin, tender meat, and rich, tomatoey sauce form a winning trifecta of delicious in this one-pot braise. We spoon it over rice and peas to soak up every last drop of goodness, and serve a tangy arugula salad alongside for a vibrant boost of flavor and color. Dinner is served! Cook, relax, and enjoy!",
  "id": "437eO3ORCME46i02SeCW46",
  "image": "//images.ctfassets.net/kk2bw5ojx476/5mFyTozvSoyE0Mqseoos86/fb88f4302cfd184492e548cde11a2555/SKU1479_Hero_077-71d8a07ff8e79abcb0e6c0ebf0f3b69c.jpg",
  "tags": Array [
    "gluten free",
    "healthy",
  ],
  "title": "Crispy Chicken and Rice	with Peas & Arugula Salad",
}
`;
